test/commands/tokens/create-test.ts

```diff
@@ -0,0 +1,155 @@
+import { expect, use } from "chai";
+import * as Nock from "nock";
+import * as ChaiAsPromised from "chai-as-promised";
+
+use(ChaiAsPromised);
+
+import CreateTokenCommand from "../../../src/commands/tokens/create";
+import { CommandArgs, CommandFailedResult, ErrorCodes } from "../../../src/util/commandline";
+
+describe("Tokens Create", () => {
+  const fakeAppOwner = "fakeAppOwner";
+  const fakeAppName = "fakeAppName";
+  const fakeAppIdentifier = `${fakeAppOwner}/${fakeAppName}`;
+  const fakeToken = "c1o3d3e7";
+  const fakeDescription = "fakeDescription";
+  /* tslint:disable-next-line:no-http-string */
+  const fakeHost = "http://localhost:1700";
+
+  let nockScope: Nock.Scope;
+
+  before(() => {
+    Nock.disableNetConnect();
+  });
+
+  beforeEach(() => {
+    nockScope = Nock(fakeHost);
+  });
+
+  afterEach(() => {
+    Nock.cleanAll();
+  });
+
+  after(() => {
+    Nock.enableNetConnect();
+  });
+
+  it("#create providing 'user' as type calls the user API", async () => {
+    const fakeId = "00000000-0000-0000-0000-000000000000";
+    const createUserTokenUrl = `/v0.1/api_tokens`;
+
+    nockScope.post(createUserTokenUrl).reply(201, {
+      id: fakeId,
+      api_token: "token",
+      scope: "all",
+    });
+
+    const command = new CreateTokenCommand(getCommandArgs(["--type", "user"]));
+    const result = await command.execute();
+
+    expect(result.succeeded).to.be.true;
+
+    nockScope.done();
+  });
+
+  it("#create user type calls user API with description when description provided", async () => {
+    const fakeId = "00000000-0000-0000-0000-000000000000";
+    const createUserTokenUrl = `/v0.1/api_tokens`;
+
+    nockScope.post(createUserTokenUrl, { description: fakeDescription }).reply(201, {
+      id: fakeId,
+      description: fakeDescription,
+      api_token: "token",
+      scope: "all",
+    });
+
+    const command = new CreateTokenCommand(getCommandArgs(["--type", "user", "--description", fakeDescription]));
+    const result = await command.execute();
+
+    expect(result.succeeded).to.be.true;
+
+    nockScope.done();
+  });
+
+  it("#create providing no args succeeds in calling the default user API", async () => {
+    const fakeId = "00000000-0000-0000-0000-000000000000";
+    const createUserTokenUrl = `/v0.1/api_tokens`;
+
+    nockScope.post(createUserTokenUrl, {}).reply(201, {
+      id: fakeId,
+      description: fakeDescription,
+      api_token: "token",
+      scope: "all",
+    });
+
+    const command = new CreateTokenCommand(getCommandArgs([]));
+    const result = await command.execute();
+
+    expect(result.succeeded).to.be.true;
```

Comment:
Maybe also check somehow that type is user? Is that possible?

Reply:
the test only mocks that the user endpoint is being called - which validates that the command defaults to user type token. I might be able to mock the app token endpoint and make sure that it's not being called, but it shouldn't be necessary to cover this case